### ISSUE TITLE
FEATURE - Small Widget, View Maintenance Events

### ISSUE THREAD

**Issue by mikaelacaron** [mikaelacaron/Basic-Car-Maintenance#15]
## Describe the feature you want the app to have that it currently doesn't
* Create a small widget that allows you to view the total number of maintenance events for a specific vehicle
   * Be able to configure which vehicle you want to view for this widget

## Describe alternatives you've considered
* N/A

## Additional context
* Only make the small widget (I'll create new issues for adding different widget sizes later

**Comment by rajhraval** [comment]
Would love to work on this issue.

**Comment by ayushs-2k4** [comment]
I can take this one, please assign it to me.

**Comment by mikaelacaron** [comment]
@rajhraval You commented first, it's yours! @ayushs-2k4 please look at another issues, thank you!

**Comment by mikaelacaron** [comment]
Notice on `dev` I created the Widget extension now, to hopefully cut down on merge conflicts

**Comment by rajhraval** [comment]
I didn't understood @mikaelacaron I would be starting to work now. Just wanted to clarify do I need to pull or do anything?

**Comment by mikaelacaron** [comment]
You will need to sync your fork see [this livestream](https://www.youtube.com/live/60ivxAOUrYU?si=GDZ7C25A7O_xsICE) at `54:35` to learn more on this process (or lookup git tutorials)

**Comment by rajhraval** [comment]
Thank you for this video explanation!

**Comment by rajhraval** [comment]
@mikaelacaron I just want to clarify the feature once again.

To create a small widget for the app, where the user can view the number of specific maintainances for a vehicle.
ie: Car Name - Total Maintenances done: 7

Currently while adding a maintainance we don't add a specific vehicle, so we need to make one.
Hope it's correct 😅

Just a suggestion, if there is a small user flow could have help better to onboard the app.
Also I think that the video you linked here, could be in the README.md it would be better for novices like me to understand things better.

**Comment by mikaelacaron** [comment]
Oh shoot sorry that's right I haven't actually connected the models together yet, I was going to do that this week, but this repo has been more popular than I was guessing and I haven't been able to do coding on my own yet. I'm going to put this on hold for right now, feel free to pick another issue, but I'll leave you as the assignee and I'll let you know when you can start on this task, if that works?

**Comment by rajhraval** [comment]
Sure, do let me know when I can start. By the way, a small flow diagram or something of the app would give a much clear picture😅

**Comment by mikaelacaron** [comment]
> Sure, do let me know when I can start. By the way, a small flow diagram or something of the app would give a much clear picture😅

Add this as an issue! This would be a great thing to add to the DocC documentation that I'm working on

**Comment by mikaelacaron** [comment]
Please wait on #141 and once you see that's merged, you can begin this issue, thanks for your patience!

**Comment by mikaelacaron** [comment]
This can be worked on now! Thanks so much for the patience

**Comment by mikaelacaron** [comment]
Actually sorry, hold off again, this isn't saving correctly

**Comment by rajhraval** [comment]
@mikaelacaron I can't work on this anymore due to some work commitments. Sorry 😅

**Comment by Jakayus** [comment]
Hey @mikaelacaron, can I get this one when/if it gets off hold?

**Comment by mikaelacaron** [comment]
@Jakayus Yes assigning it to you now! I'll try to work on the fix for this tonight

**Comment by mikaelacaron** [comment]
Thanks for your patience! I'm going to hold off on all the widget tasks for right now because I'm going to change how the vehicles and maintenance events are connected. But because it's so close to the end of Hacktoberfest I don't want to break everything for everyone, so I'll open this issue again after Hacktoberfest and you're welcome to contribute to it for fun if you'd like

**Comment by Jakayus** [comment]
No problem! I'll keep an eye out after Hacktoberfest, it seems like a fun Issue to tackle

**Comment by mikaelacaron** [comment]
Opening this back up for Hacktoberfest 2024!

**Comment by anupdsouza** [comment]
@mikaelacaron, can I look into this one?

**Comment by anupdsouza** [comment]
@mikaelacaron, I'm un-assigning  myself as I'm occupied with other things & haven't been able to make time for this. Sorry!

**Comment by mikaelacaron** [comment]
No problem! Thanks for un-assigning yourself!

**Comment by narwalayush4** [comment]
@mikaelacaron I would like to work on this, as #315 is a no go.

**Comment by mikaelacaron** [comment]
You got it!

**Comment by TMaszko** [comment]
@mikaelacaron Hi! As I'm almost total newbie in iOS world I'd love to pick this one up ;)

**Comment by mikaelacaron** [comment]
Hey @TMaszko go for it!! please feel free to ask any questions! Even if you feel like they're dumb questions, because we can all learn something new

**Comment by TMaszko** [comment]
@mikaelacaron First PR in Swift! so please keep that in mind ;) #394
